This patch has been checked into Scintilla CVS - so it can be removed on the
next Scintilla update.

Fix dedenting a rectangular selection using Shift+Tab (bug 86170).

Index: /home/toddw/src/editors/scintilla/cvs/scintilla/src/Editor.cxx
===================================================================
--- src/Editor.cxx	6 Mar 2010 09:50:03 -0000	1.515
+++ src/Editor.cxx	8 Mar 2010 23:39:26 -0000
@@ -5215,7 +5215,7 @@
 					int indentation = pdoc->GetLineIndentation(lineCurrentPos);
 					int indentationStep = pdoc->IndentSize();
 					pdoc->SetLineIndentation(lineCurrentPos, indentation - indentationStep);
-					SetEmptySelection(pdoc->GetLineIndentPosition(lineCurrentPos));
+					sel.Range(r) = SelectionRange(pdoc->GetLineIndentPosition(lineCurrentPos));
 				} else {
 					int newColumn = ((pdoc->GetColumn(caretPosition) - 1) / pdoc->tabInChars) *
 							pdoc->tabInChars;
